core/shared/src/main/scala/zio/prelude/Traversable.scala

```diff
@@ -269,17 +270,68 @@ trait Traversable[F[+_]] extends Covariant[F] {
 
 object Traversable extends LawfulF.Covariant[DeriveEqualTraversable, Equal] {
 
+  /**
+   * Identity Law :
+   *  traverse Identity ta = Identity ta
+   */
+  val traversableIdentityLaw: LawsF.Covariant[DeriveEqualTraversable, Equal] =
+    new LawsF.Covariant.Law1[DeriveEqualTraversable, Equal]("traversableIdentityLaw") {
+      def apply[F[+_]: DeriveEqualTraversable, A: Equal](fa: F[A]): TestResult =
+        fa.foreach(Id(_)) <-> Id(fa)
+    }
+
+  /**
+   *  Composition Law for various kind of Applivatives
+   */
+  val traversableCompositionLaw: LawsF.Covariant[DeriveEqualTraversable, Equal] = {
+    compositionLawCase[Id, Id] + compositionLawCase[Option, List] + compositionLawCase[List, Option]
+  }
+
+  /**
+   * Composition law
+   * traverse (Compose . fmap g . f) ta = Compose . fmap (traverse g) . traverse f $ ta
```

Comment:
I would be worried that the Haskell names could confuse some users/contributors. I think we should use the ZIO Prelude terminology here, so `foreach` (instead of `traverse`) and `map` (instead of `fmap`).